google/resource_google_folder_iam_policy_test.go

```diff
@@ -0,0 +1,168 @@
+		Steps: []resource.TestStep{
+			resource.TestStep{
+				Config: testAccGoogleFolderIamPolicy_basic(folderDisplayName, parent, policy),
+				Check:  testAccCheckGoogleFolderIamPolicy("google_folder_iam_policy.test", policy),
+			},
+		},
+	})
+}
+
+func TestAccGoogleFolderIamPolicy_update(t *testing.T) {
+	skipIfEnvNotSet(t, "GOOGLE_ORG")
+
+	folderDisplayName := "tf-test-" + acctest.RandString(10)
+	org := os.Getenv("GOOGLE_ORG")
+	parent := "organizations/" + org
+
+	policy1 := &resourceManagerV2Beta1.Policy{
+		Bindings: []*resourceManagerV2Beta1.Binding{
+			{
+				Role: "roles/viewer",
+				Members: []string{
+					"user:admin@hashicorptest.com",
+				},
+			},
+		},
+	}
+	policy2 := &resourceManagerV2Beta1.Policy{
+		Bindings: []*resourceManagerV2Beta1.Binding{
+			{
+				Role: "roles/editor",
+				Members: []string{
+					"user:admin@hashicorptest.com",
+				},
+			},
+			{
+				Role: "roles/viewer",
+				Members: []string{
+					"user:admin@hashicorptest.com",
+				},
+			},
+		},
+	}
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckGoogleFolderIamPolicyDestroy,
+		Steps: []resource.TestStep{
+			resource.TestStep{
+				Config: testAccGoogleFolderIamPolicy_basic(folderDisplayName, parent, policy1),
+				Check:  testAccCheckGoogleFolderIamPolicy("google_folder_iam_policy.test", policy1),
+			},
+			resource.TestStep{
+				Config: testAccGoogleFolderIamPolicy_basic(folderDisplayName, parent, policy2),
+				Check:  testAccCheckGoogleFolderIamPolicy("google_folder_iam_policy.test", policy2),
+			},
+		},
+	})
+}
+
+func testAccCheckGoogleFolderIamPolicyDestroy(s *terraform.State) error {
+	config := testAccProvider.Meta().(*Config)
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "google_folder_iam_policy" {
+			continue
+		}
+
+		folder := rs.Primary.Attributes["folder"]
+		policy, err := config.clientResourceManagerV2Beta1.Folders.GetIamPolicy(folder, &resourceManagerV2Beta1.GetIamPolicyRequest{}).Do()
+
+		if err != nil && len(policy.Bindings) > 0 {
+			return fmt.Errorf("Folder '%s' policy hasn't been deleted.", folder)
+		}
+	}
+	return nil
+}
+
+func testAccCheckGoogleFolderIamPolicy(n string, policy *resourceManagerV2Beta1.Policy) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[n]
+		if !ok {
+			return fmt.Errorf("Not found: %s", n)
+		}
+
+		if rs.Primary.ID == "" {
+			return fmt.Errorf("No ID is set")
+		}
+
+		config := testAccProvider.Meta().(*Config)
+
+		p, err := config.clientResourceManagerV2Beta1.Folders.GetIamPolicy(rs.Primary.ID, &resourceManagerV2Beta1.GetIamPolicyRequest{}).Do()
+		if err != nil {
+			return err
+		}
+
+		if !reflect.DeepEqual(p.Bindings, policy.Bindings) {
+			return fmt.Errorf("Incorrect iam policy bindings. Expected '%s', got '%s'", policy.Bindings, p.Bindings)
+		}
+
```

Comment:
Also needs a check to make sure etag was stored

Reply:
Done